sdk/iOS/src/MSSyncContext.m

```diff
@@ -404,54 +406,53 @@ - (void) pullWithQuery:(MSQuery *)query queryId:(NSString *)queryId completion:(
-        // Before we can pull from the remote, we need to make sure out table doesn't having pending operations
-        NSArray *tableOps = [self.operationQueue getOperationsForTable:query.table.name item:nil];
-        if (tableOps.count > 0) {
-            [self pushWithCompletion:^(NSError *error) {
-                // For now we just abort the pull if the push failed to complete successfully
-                // Long term we can be smarter and check if our table succeeded
-                if (error) {
-                    if (completion) {
-                        [self.callbackQueue addOperationWithBlock:^{
-                            completion(error);
-                        }];
-                    }
-                } else {
-                    // Check again if we have new pending ops while we synced, and repeat as needed
-                    [self pullWithQueryInternal:query queryId:queryId maxRecords:maxRecords completion:completion];
-                }
-            }];
-            return;
-        } else {
-            // TODO: Allow users to cancel operations
-            MSQueuePullOperation *pull = [[MSQueuePullOperation alloc] initWithSyncContext:self
-                                                                                     query:query
-                                                                                   queryId:queryId
-                                                                                maxRecords:maxRecords
-                                                                             dispatchQueue:writeOperationQueue
-                                                                             callbackQueue:self.callbackQueue
-                                                                                completion:completion];
-            
-            
-            [pushQueue_ addOperation:pull];
-        }
-    });
+	MSQueuePullOperation *pull = [[MSQueuePullOperation alloc] initWithSyncContext:self
+																			 query:query
+																		   queryId:queryId
+																		maxRecords:maxRecords
+																	 dispatchQueue:writeOperationQueue
+																	 callbackQueue:self.callbackQueue
+																		completion:completion];
+	
+	// Before we can pull from the remote, we need to make sure out table doesn't having pending operations
+	NSArray *tableOps = [self.operationQueue getOperationsForTable:query.table.name item:nil];
+	if (tableOps.count > 0) {
+		MSQueuePushOperation *push = [self pushWithCompletion:^(NSError *error) {
+			// For now we just abort the pull if the push failed to complete successfully
+			// Long term we can be smarter and check if our table succeeded
+			if (error) {
+				if (completion) {
+					[self.callbackQueue addOperationWithBlock:^{
+						completion(error);
+					}];
+				}
+			} else {
+				// Check again if we have new pending ops while we synced, and repeat as needed
+				[pushQueue_ addOperation:pull];
```

Comment:
I couldn't figure out how to recheck if the table was dirty and still utilise the pull operation we returned to the user. 

The only was I cam up with was extracting this dirty/push check into another private method which took the existing pull operation. Thus we could call it again in the callback and pass the contained push operation. Thoughts?